File: railway-backend/src/models/cityAlertModels.ts
export type AlertCategory = 'WEATHER' | 'DISASTER' | 'AIR_QUALITY';

export type AlertSeverity = 'INFO' | 'LOW' | 'MODERATE' | 'HIGH' | 'SEVERE' | 'CRITICAL';

export type AlertStatus = 'ACTIVE' | 'UPCOMING' | 'EXPIRED';

export interface CityAlert {
  id: string;
  type?: string;
  title: string;
  description: string;
  category: AlertCategory;
  severity: AlertSeverity;
  status: AlertStatus;
  startTime?: string;
  endTime?: string;
  lastUpdated?: string;
  locationName?: string;
  city?: string;
  district?: string;
  latitude?: number;
  longitude?: number;
  sourceName: string;
  sourceUrl?: string;
  isVerified: boolean;
}

export interface CityAlertsResponse {
  status: string;
  location?: {
    latitude: number;
    longitude: number;
    city?: string;
    district?: string;
  };
  alerts: CityAlert[];
  updatedAt: string;
}

File: app/src/main/java/com/example/smartcityassistant/data/cityalerts/CityAlertModels.kt
package com.example.smartcityassistant.data.cityalerts

data class CityAlert(
    val id: String,
    val type: String?,
    val title: String,
    val description: String,
    val category: String,
    val severity: String,
    val status: String,
    val startTime: String?,
    val endTime: String?,
    val lastUpdated: String?,
    val locationName: String?,
    val city: String? = null,
    val district: String? = null,
    val latitude: Double?,
    val longitude: Double?,
    val sourceName: String,
    val sourceUrl: String?,
    val isVerified: Boolean
)

data class CityAlertLocation(
    val latitude: Double,
    val longitude: Double,
    val city: String?,
    val district: String?
)

data class CityAlertsResponse(
    val status: String,
    val location: CityAlertLocation?,
    val alerts: List<CityAlert>,
    val updatedAt: String?
)
